### DISCUSSION TITLE
Quick Autocomplete for Buy Activities

### DISCUSSION THREAD

**Discussion by Usiel** [ghostfolio/ghostfolio#3421] [Ideas]
When adding new activities I often add stock/symbols I added previously. Unfortunately, when I add them by name it can be quite cumbersome to find the stock by name, especially when they come in different variants (currencies, H, acc, ...), as I have to wait for the autocomplete to load (the naming my broker/bank uses may be different to Yahoo).

I was wondering if there's demand for a quick autocomplete based on existing activities? The user would see autocomplete suggestions based on their activities almost immediately (no external calls) and then a bit later they would see the suggestions from external sources. Local suggestions would always be on top. The local suggestions could even be using 2 sources, a truly local and instant (as in the 50 activities already loaded client-side) source and a Ghostfolio server source that combs through the last X activities.

**Comment by dtslvr** [comment]
Thanks for sharing your idea, @Usiel.

Have you already tried the clone functionality? It may be helpful until autocompletion is available in Ghostfolio.

**Reply by Usiel** [reply]
Thanks, clone is a good fit for this use-case.

I may still look into the autocomplete enhancement if I get some free time - assuming there's demand.

**Reply by dtslvr** [reply]
What do you think about reusing the endpoint `api/v1/portfolio/holdings?query=...` for highlighting the symbols already in use?

I find it more cumbersome to keep the data locally than to retrieve them from an existing endpoint.

**Reply by Usiel** [reply]
> I find it more cumbersome to keep the data locally than to retrieve them from an existing endpoint.

True, also it would make the autocomplete (and the whole "Add activity" form) independent of the location.

> What do you think about reusing the endpoint `api/v1/portfolio/holdings?query=...` for highlighting the symbols already in use?

Yeah, that sounds good to me as a first version. If we need extra performance we could consider implementing more field projection options in the existing endpoint or adding a separate endpoint just for autocomplete (returning only the required fields, not querying anything on the DB beyond that).

I can look into it.

**Reply by Usiel** [reply]
Finally got around to doing this and testing it properly during my monthly data input :) See https://github.com/ghostfolio/ghostfolio/pull/3507 for a prototype.
